src/streamlink_cli/argparser.py

```diff
@@ -553,6 +553,15 @@ def build_parser():
         You will be prompted if the file already exists.
         """
     )
+    output.add_argument(
+        "--mux-subtitles",
+        action="store_true",
+        help="""
+        Automatically mux available subtitles in to the output stream.
+
+        If supported by the plugin.
+        """,
+    )
```

Comment:
The parameter belongs into the `transport` group, I think.

You'll also have to add it to `streamlink_cli.main.setup_options` and `streamlink.plugin.plugin`. In `streamlink.plugin.plugin`, where the `self.session.get_option` call reads from, the parameter name needs to include a dash and not an underscore character. This needs to be changed in each plugin.

Reply:
I figured as it affects the output it could go in there, but maybe transport makes sense.

Reply:
re:decorator, that looks like a nice idea - but maybe it could be applied to the plugin and that information could be picked up in help somewhere. There is perhaps the possibility for something generic here too.

Reply:
The more I look at it, I think that there is something we could do similar to how `PluginArguments` works for plugins, but for Streamlink that way other interfaces could interrogate streamlink to find out which options are supported.